components/ws-manager-bridge/src/cluster-service-server.ts

```diff
@@ -378,6 +380,23 @@ export class ClusterServiceServer {
         }
     }
 
+    public async getScore(){
+        // TODO(arthur): findFiltered needs a proper filter.
+        // How can I filter with "The cluster where this component is running"?
+        for (const cluster of await this.clusterDB.findFiltered({})) {
+            return new Promise<number>((resolve) => {
+                const clusterStatus = convertToGRPC(cluster);
+                resolve(clusterStatus.getScore())
+            });
+        }
```

Comment:
I'mm getting an error here:
```
(node:30) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'findFiltered' of undefined\n    at ClusterServiceServer.<anonymous> (/app/node_modules/@gitpod/ws-manager-bridge/dist/src/cluster-service-server.js:528:61)\n    at step (/app/node_modules/@gitpod/ws-manager-bridge/dist/src/cluster-service-server.js:60:23)\n    at Object.next (/app/node_modules/@gitpod/ws-manager-bridge/dist/src/cluster-service-server.js:41:53)\n    at /app/node_modules/@gitpod/ws-manager-bridge/dist/src/cluster-service-server.js:35:71\n    at new Promise (<anonymous>)\n    at __awaiter (/app/node_modules/@gitpod/ws-manager-bridge/dist/src/cluster-service-server.js:31:12)\n    at ClusterServiceServer.getScore (/app/node_modules/@gitpod/ws-manager-bridge/dist/src/cluster-service-server.js:513:16)\n    at /app/node_modules/@gitpod/ws-manager-bridge/dist/src/main.js:84:52\n    at new Promise (<anonymous>)\n    at /app/node_modules/@gitpod/ws-manager-bridge/dist/src/main.js:83:21\n    at Layer.handle [as handle_request] (/app/node_modules/express/lib/router/layer.js:95:5)\n    at next (/app/node_modules/express/lib/router/route.js:137:13)\n    at Route.dispatch (/app/node_modules/express/lib/router/route.js:112:3)\n    at Layer.handle [as handle_request] (/app/node_modules/express/lib/router/layer.js:95:5)\n    at /app/node_modules/express/lib/router/index.js:281:22\n    at Function.process_params (/app/node_modules/express/lib/router/index.js:335:12)
```

Which makes sense, since I'm not declaring this `clusterDB` object anywhere. Could anyone provide any tips on how to make this work?

Reply:
I think you need an inject (for the dependency injection) as here:

https://github.com/gitpod-io/gitpod/blob/66aa4bd6c2f9299df8429ebc4f4179e75de55aa7/components/ws-manager-bridge/src/cluster-service-server.ts#L49-L50